nixos/modules/services/networking/netmaker/netmaker.nix

```
@@ -0,0 +1,787 @@
+        - `''${services.netmaker.coredns.networks.domain}` (defaults to: `int.` prefix)
+        - `ns.''${services.netmaker.domain}` - just a suggestion
+
+        You might want to create following DNS records at your root DNS zone registrar before applying the module:
+
+            api.<YOUR-DOMAIN>.         1       IN      A       <YOUR-SERVER-IP>
+            broker.<YOUR-DOMAIN>.      1       IN      A       <YOUR-SERVER-IP>
+            dashboard.<YOUR-DOMAIN>.   1       IN      A       <YOUR-SERVER-IP>
+            ns.<YOUR-DOMAIN>.          1       IN      A       <YOUR-SERVER-IP>
+            int.<YOUR-DOMAIN>.         1       IN      NS      ns.<YOUR-DOMAIN>
+      '';
+    };
+
+    email = lib.mkOption {
+      type = with lib.types; str;
+      description = lib.mdDoc ''
+        Email address to (currently) use for certificato registration.
+      '';
+    };
+
+    coredns.networks.domain = lib.mkOption {
+      type = with lib.types; str;
+      default = "int.${config.services.netmaker.domain}";
+      defaultText = lib.literalExpression ''"int.''${config.services.netmaker.domain}"'';
+      description = lib.mdDoc ''
+        The domain CoreDNS (by default) exposes Netmaker's networks under.
+      '';
+    };
+    api.domain = lib.mkOption {
+      type = with lib.types; str;
+      default = "api.${config.services.netmaker.domain}";
+      defaultText = lib.literalExpression ''"api.''${config.services.netmaker.domain}"'';
+      description = lib.mdDoc ''
+        The domain Netmaker Server's API is exposed on.
+      '';
+    };
+    ui.domain = lib.mkOption {
+      type = with lib.types; str;
+      default = "dashboard.${config.services.netmaker.domain}";
+      defaultText = lib.literalExpression ''"dashboard.''${config.services.netmaker.domain}"'';
+      description = lib.mdDoc ''
+        The domain Netmaker UI is exposed on.
+      '';
+    };
+    mqtt.domain = lib.mkOption {
+      type = with lib.types; str;
+      default = "broker.${cfg.domain}";
+      defaultText = lib.literalExpression ''"broker.''${config.services.netmaker.domain}"'';
+      description = lib.mdDoc ''
+        Domain the message queue is proxied through for external access.
+      '';
+    };
+
+    package = lib.mkOption {
+      type = with lib.types; package;
+      default = if cfg.installType == "ce" then pkgs.netmaker else pkgs.netmaker-pro;
+      defaultText = lib.literalExpression ''if cfg.installType == "ce" then pkgs.netmaker else pkgs.netmaker-pro'';
+      description = lib.mdDoc ''
+        Package providing `netmaker` binary.
+      '';
+    };
+
+    nmctl.package = lib.mkOption {
+      type = with lib.types; package;
+      default = pkgs.nmctl;
+      defaultText = lib.literalExpression ''pkgs.nmctl'';
+      description = lib.mdDoc ''
+        Package providing `nmctl` binary.
+      '';
+    };
+
+    internalListenAddress = lib.mkOption {
+      type = with lib.types; str;
```

```suggestion
      type = lib.types.str;
```
we shouldn't use with if it is not necessary